dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -2,203 +2,277 @@
 import com.dotcms.enterprise.license.LicenseLevel;
 import com.dotcms.enterprise.publishing.bundlers.FileAssetBundler;
 import com.dotcms.enterprise.publishing.bundlers.HTMLPageAsContentBundler;
 import com.dotcms.enterprise.publishing.bundlers.URLMapBundler;
 import com.dotcms.enterprise.publishing.sitesearch.SiteSearchConfig;
+import com.dotcms.enterprise.publishing.sitesearch.SiteSearchResults;
 import com.dotcms.publishing.BundlerStatus;
 import com.dotcms.publishing.DotPublishingException;
 import com.dotcms.publishing.PublishStatus;
+import com.dotcms.publishing.PublisherAPI;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.UserAPI;
 import com.dotmarketing.db.HibernateUtil;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
 import com.dotmarketing.sitesearch.business.SiteSearchAPI;
+import com.dotmarketing.sitesearch.business.SiteSearchAuditAPI;
 import com.dotmarketing.sitesearch.model.SiteSearchAudit;
-import com.dotmarketing.util.*;
+import com.dotmarketing.util.ActivityLogger;
+import com.dotmarketing.util.AdminLogger;
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.StringUtils;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.annotations.VisibleForTesting;
 import com.liferay.portal.model.User;
-import org.elasticsearch.ElasticsearchException;
-import org.quartz.JobDataMap;
-import org.quartz.JobExecutionContext;
-import org.quartz.JobExecutionException;
-
+import com.liferay.util.StringPool;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.Date;
 import java.util.List;
+import java.util.ListIterator;
 import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.BooleanUtils;
+import org.elasticsearch.ElasticsearchException;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
 
 public class SiteSearchJobImpl {
+
+    static final String INCREMENTAL = "incremental";
+    static final String LANG_TO_INDEX = "langToIndex";
+    static final String INDEX_ALIAS = "indexAlias";
+    static final String INDEX_ALL = "indexAll";
+    static final String INDEX_HOST = "indexhost";
+    static final String QUARTZ_JOB_NAME = "QUARTZ_JOB_NAME";
+    static final String RUNNING_ONCE_JOB_NAME = "runningOnce";
+    static final String JOB_ID = "JOB_ID";
+    static final String RUN_NOW = "RUN_NOW";
+    static final String INCLUDE_EXCLUDE = "includeExclude";
+    static final String INCLUDE = "include";
+    static final String PATHS = "paths";
+
+    private final ESIndexAPI esIndexAPI;
+    private final IndiciesAPI indicesAPI;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'indicesAPI' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093862)